packages/experiments/src/components/Recommendation/Recommendation.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+
+/* Dependent Components */
+import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import { CardFrame, ICardDropDownOption } from '../Card/CardFrame';
+
+/* Types for props and styles */
+import { IRecommendationProps, IRecommendationStyles } from './Recommendation.types';
+
+/* Styles for CardComponent and Recommendation Card */
+import { CardComponentStyles, getStyles } from './Recommendation.styles';
+
+/* Utilities */
+import { classNamesFunction } from 'office-ui-fabric-react/lib/Utilities';
+
+export class Recommendation extends React.Component<IRecommendationProps, {}> {
+  /* Default Props */
+  public static defaultProps = {
+    recommendationButtonLocalizedName: 'View Recommendation',
+    recommendationButtonAriaDescription: 'Click to View Recommendation',
+    dismissRecommendationLocalizedName: 'Dismiss',
+    dismissRecommendationAriaLabel: 'Dismiss Recommendation'
+  };
+
+  private recommendationMenuItems: ICardDropDownOption[];
+
+  constructor(props: IRecommendationProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const {
+      recommendationBarTitle,
+      recommendationDescriptionHeader,
+      recommendationDescription,
+      recommendationButtonLocalizedName,
+      recommendationButtonAriaDescription,
+      handleViewRecommendationClick,
+      dismissRecommendationLocalizedName,
+      dismissRecommendationAriaLabel,
+      handleDismissRecommendationClick
+    } = this.props;
+
+    const getClassNames = classNamesFunction<{}, IRecommendationStyles>();
```

Comment:
>const getClassNames = classNamesFunction<{}, IRecommendationStyles>(); [](start = 4, length = 70)

can be moved outside the component... why is it inside render ?

Reply:
Good call, move it outside of the component.

---
In reply to: [196373280](https://github.com/OfficeDev/office-ui-fabric-react/pull/5252#discussion_r196373280) [](ancestors = 196373280)